src/main/scala/com/fulcrumgenomics/illumina/Sample.scala

```diff
@@ -42,22 +46,30 @@ package com.fulcrumgenomics.util.miseq
   * @param lane the lane number for the sample.
   * @param i7IndexBases the sample barcode bases in the i7 read.
   * @param i5IndexBases the sample barcode bases in the i5 read.
-  * @param sampleBarcode the full sample barcode for optimized access.
   * @param extendedAttributes a map of non-standard or site-specific extended attributes. Keys should be upper-case.
   */
-class Sample(val sampleOrdinal: Int,
-             val sampleId: String,
-             val sampleName: String,
-             val libraryId: Option[String] = None,
-             val project: Option[String] = None,
-             val description: Option[String] = None,
-             val lane: Option[Int] = None,
-             val i7IndexBases: Option[String] = None,
-             val i5IndexBases: Option[String] = None,
-             var sampleBarcode: Option[SampleBarcode] = None,
-             val extendedAttributes: Map[String, String] = Map.empty) {
+case class Sample(sampleOrdinal: Int,
+                  sampleId: String,
+                  sampleName: String,
+                  libraryId: String,
+                  project: Option[String] = None,
+                  description: Option[String] = None,
+                  lane: Option[Int] = None,
+                  i7IndexBases: Option[String] = None,
+                  i5IndexBases: Option[String] = None,
+                  extendedAttributes: Map[String, String] = Map.empty) {
   extendedAttributes.keysIterator.foreach { key => require(key == key.toUpperCase) }
 
+  import Sample.SampleBarcodeDelimiter
+
+  def sampleBarcodes: Seq[Option[String]] = Seq(i7IndexBases, i5IndexBases)
+
+  lazy val sampleBarcodeBytes: Array[Byte]  = sampleBarcodes.flatten.filter(_.nonEmpty).flatMap(_.getBytes).toArray
+  lazy val sampleBarcodeString: String      = sampleBarcodes.flatten.filter(_.nonEmpty).mkString(SampleBarcodeDelimiter)
+  override def equals(o: Any): Boolean = o.isInstanceOf[Sample] && this.sampleBarcodeString == o.asInstanceOf[Sample].sampleBarcodeString
```

Comment:
Copy and past error from the previous `SampleBarcode` class.  Thanks for catching this.